onward/app/controllers/MostPopularController.scala

```diff
@@ -200,6 +203,47 @@ class MostPopularController(
       }
     }
 
+  def renderWithDeeplyRead(): Action[AnyContent] =
+    Action.async { implicit request =>
+      val edition = Edition(request)
+
+      // Synchronous edition popular, from the mostPopularAgent (stateful)
+      val editionPopular: Option[MostPopularTrails] = {
+        val editionPopularContent = mostPopularAgent.mostPopular(edition)
+        if (editionPopularContent.isEmpty) None
+        Some(
+          MostPopularTrails(
+            "Most popular",
+            "",
+            editionPopularContent
+              .map(_.faciaContent)
+              .map(Trail.pressedContentToTrail),
+          ),
+        )
+      }
+
+      val deeplyReadItems = deeplyReadAgent.getTrails
+
+      // Async global deeply read
+      val deeplyRead: Option[MostPopularTrails] = {
+        if (deeplyReadItems.isEmpty) None
+        Some(
+          MostPopularTrails(
+            "Deeply read",
+            "",
+            deeplyReadItems,
+          ),
+        )
+      }
+
+      val response =
+        if (editionPopular.isDefined && deeplyRead.isDefined)
+          jsonResponseTrails(editionPopular.toSeq ++ deeplyRead.toSeq, mostCards())
+        else NotFound
```

Comment:
One aspect that is puzzling me is that I have seen responses with empty arrays when first starting the app—including in our pairing session—but the logic is supposed to return a `NotFound` response in case either of the two most popular trail collections is empty. 🤔 


We could use pattern matching here, if that helps readability? Not sure…


```suggestion
      val response = (editionPopular, deeplyRead) match => {
        case (Some(p), Some(d)) => 
          jsonResponseTrails(p.toSeq ++ d.toSeq, mostCards())
        case (_, _) => NotFound
      }
```